Crystalball/filtering.py

```diff
@@ -0,0 +1,75 @@
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def valid_field_ids(field_datasets, fields=None):
+    """
+    Parameters
+    ----------
+    field_datasets : list of :class:`daskms.Dataset`
+
+    fields : None or str
+        if None, all fields are valid, otherwise
+        the string will be split on commas and used to
+        select fields. Both field ID's and names are supported
+        in the string.
+
+        .. code-block:: python
+
+            valid_field_ids(field_ds, ["PKS-1934,1,DEEP2,3"])
+
+    Returns
+    -------
+    fields_ids : list of ints
+        List of valid field IDs
+    """
+    if fields is None:
+        return list(range(len(field_datasets)))
+    elif not isinstance(fields, str):
+        raise TypeError("fields must be None or str")
+
+    field_names = {str(ds.NAME.values): i for i, ds
+                   in enumerate(field_datasets)}
+
+    selected_fields = [f.strip() for f in fields.split(',')]
+
+    # Add integral lookup values
+    for field_name, field_id in list(field_names.items()):
+        exists = field_names.setdefault(str(field_id), field_id)
+
+        if not exists == field_id:
+            raise ValueError("Existing field_id mismatch "
+                             "for field %s. %d != %d"
+                             % (field_name, exists, field_id))
+
+    field_ids = []
+
```

Comment:
@KshitijT Can you put a

```python
from pprint import pprint
pprint(field_names)
```

here and see what happens?

Reply:
I could.

Reply:
This is what I get:
```
crystalball -sm Cyg_A_Vanilla_selfcal_ABCD-ph2-sources.txt -w 'core.reg' -sp -o FLUFFY_CAT_COLUMN -ns 1 -f 'J1331+3030' ../msdir/S-CYG-A-3.5-10S-4M.MS/ 
Successful read/write open of default-locked table ../msdir/S-CYG-A-3.5-10S-4M.MS/: 33 columns, 6305000 rows
-------------------------------------------
system RAM = 503.81 GB
nr of logical CPUs = 72
nr sources = 1
nr rows    = 6305000
nr chans   = 32
nr corrs   = 4
sources per chunk = 1 (auto setting)
rows per chunk    = 13543 (auto setting)
expected memory usage = 1.86 GB
['J1331+3030']
{'0': 0,
 '1': 1,
 '2': 2,
 "['Cygnus-Nuc']": 0,
 "['J1331+3030']": 1,
 "['J2007+4029']": 2}
Field J1331+3030 does not exist and ignored for selection

```